CMakeLists.txt

```diff
@@ -368,6 +368,15 @@ if (BUILD_TESTS)
   include(CTest)
   enable_testing()
   message(STATUS "Unit tests are enabled")
+
+  # If CTEST_PARALLEL_LEVEL is not defined, lets define it to ncpus to keep current behavior
+  execute_process(COMMAND "nproc" OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE CORES)
+  if (NOT TEST_JOB_COUNT)
+    set(TEST_JOB_COUNT ${CORES})
+  endif()
+  message(STATUS "Running tests with ${TEST_JOB_COUNT} jobs")
+  # Set TEST_JOB_FLAG to -j$TEST_JOB_COUNT
+  set(TEST_JOB_FLAG "-j${TEST_JOB_COUNT}")
```

Comment:
Any reason for not wrapping this in `if (${CMAKE_VERSION} VERSION_LESS "3.29")` to enable using `CTEST_PARALLEL_LEVEL` on newer CMake versions?

The difference doesn't matter much in our current setup, but technically the build machine could be a different one than the one running ctest, which `nproc` cannot predict.

Reply:
What's the advantage of `CTEST_PARALLEL_LEVEL`?

Reply:
`CTEST_PARALLEL_LEVEL` isn't the important part, but rather it's to make `ctest` infer the core count itself when `TEST_JOB_COUNT` is not set. Alternatively, we could default `TEST_JOB_COUNT` to 0 for CMake 3.29 instead of calling `nproc`.

Having `ctest` infer parallelization itself is better than guessing the core count at generation-time, since the latter can produce incorrect results (e.g. if the tests run on a different machine than the build host, or if hardware is upgraded). Neither of these are critical things to be concerned with, but it's equally simple to add a version check to address them.

Reply:
... indeed, if we leave `TEST_JOB_COUNT` untouched on CMake 3.29+, `TEST_JOB_FLAG` will end up as `-j`, which would also have the desired behavior. We just need to change `else()` to `elif(version check)` for this.

Reply:
Thanks!